fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -921,6 +921,27 @@ private void analyzeAddPartition(OlapTable olapTable, List<PartitionDesc> partit
         }
     }
 
+    private int calAvgBucketNumOfLastFivePartitions(OlapTable olapTable) {
+        int avgBucketNum = 0;
+        if (olapTable.getPartitions().size() < 5) {
+            avgBucketNum = GlobalStateMgr.getCurrentSystemInfo().getAvailableBackendIds().size();
```

Comment:
We need to make sure the performance will not decrease in the most typical seneria, eg: `3 BE for PoC or Benchmarking`